### PR TITLE
brp-python-bytecompile: Ensure reproducibility of invalidation timestamp

### PR DESCRIPTION
**Pull request by martyone**

### CODE REVIEW

File: rpm/0022-brp-python-bytecompile-Ensure-reproducibility-of-inv.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
++++ b/scripts/brp-python-bytecompile
+@@ -61,6 +61,13 @@ EOF
+ # This helps to make byte-compilation more reproducible
+ export PYTHONHASHSEED=0
+ 
++# .pyc/.pyo files embed the time of modification of the source file. Ensure
++# build reproducibility by adjusting the modification time a reproducible way.
++if [ -n "$SOURCE_DATE_EPOCH" ]; then
++	echo "Updating modification time of .py files according to \$SOURCE_DATE_EPOCH"
++	find "$RPM_BUILD_ROOT" -name '*.py' -print0 |xargs --verbose -0 touch --date=@"$SOURCE_DATE_EPOCH"
```
Comment (Thaodan) [inline rpm/0022-brp-python-bytecompile-Ensure-reproducibility-of-inv.patch:23]: -exec instead of xargs would work here too.

**Review by Thaodan — APPROVED** [review]
LGTM, just wondering if we should use this as the script is already gone in upstream packaging and move to the version provided by [python-rpm-macros](https://github.com/sailfishos-mirror/python-rpm-macros) or python-rpm-packaging.

**Review by vigejolla — APPROVED** [review]
LGTM

File: rpm/0022-brp-python-bytecompile-Ensure-reproducibility-of-inv.patch (hunk 1 above)
Comment (martyone) [reply rpm/0022-brp-python-bytecompile-Ensure-reproducibility-of-inv.patch:23]: `-exec` would spawn `touch` for each file, `xargs` spawns once for `ARG_MAX`.

**Comment by martyone** [comment]
> I wondering if we should use this as the script is already gone in upstream packaging and move to the version provided by [python-rpm-macros](https://github.com/sailfishos-mirror/python-rpm-macros) or python-rpm-packaging.

It would make more sense to me to do such move in scope of rpm update instead.

**Pull request merged**